File: web/src/lib/crypto/kyber.ts
// hello OS v2.0 — Kyber KEM Types & Stub
// Re-exports KEM interface and stub implementations from pqxdh.ts.
// When a production Kyber-1024 WASM/JS library is available,
// add the real implementation here alongside the stub.

export type { KemAlgorithm, KemKeyPair, KemEncapsulation } from './pqxdh';
export { StubKyber, DeterministicStubKyber } from './pqxdh';

File: engine/lib/src/crypto/pqxdh/kyber.dart
/// Kyber KEM types — re-export barrel for PQXDH.
///
/// NOTE: The concrete `StubKyber` and `DeterministicStubKyber` classes
/// are **defined in `pqxdh.dart`**, not here. This file exists solely
/// to provide a stable `kyber.dart` import path for callers. When a
/// production Kyber-1024 library becomes available in pure Dart, add
/// the real implementation either inline here or alongside the stubs
/// in `pqxdh.dart` and export it from this barrel.
library;

export 'pqxdh.dart'
    show
        KemAlgorithm,
        KemKeyPair,
        KemEncapsulation,
        StubKyber,
        DeterministicStubKyber;
